src/button/button-group.scss

```diff
@@ -10,16 +10,22 @@
   $childSelector: 'input, button'; // target nested inputs and buttons
 
   > * {
+    display: flex;
+
     #{$childSelector} {
-      &:hover,
-      &:focus,
-      &:focus-within {
-        position: relative;
+      &:hover {
         z-index: 1;
       }
 
+      &:focus {
+        z-index: 2;
+
+        &:where(:not(:focus-visible):not(:hover)) {
```

Comment:
Is `where` needed?

Reply:
Yes, we want to keep the specificity low and use the cascade. `where` has similar browser support as `focus-visible` so it's not an issue.